### PR TITLE
New Source : OpenSearch connector

### PR DESCRIPTION
**Pull request by jay-choe**
Hi team, Our team want to use OpenSearch as a source, so we are gonna add new OpenSearch Connector based on java.
Feel free to comment our source and let us know if there is any problem.
You guys closed last PR and suggested reopen it so we did it.
Check it plz and feel free to let us know to fix if needed.

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
I tested the integration test and its not working. Can you confirm you can run locally? 
`./gradlew airbyte-integrations:connectors:source-opensearch:integrationTest`

Also it's missing the icon and the metadata.yaml file inside the connector folder. Take a look in other Java connectors to reference.

File: airbyte-integrations/connectors/destination-opensearch/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM airbyte/integration-base-java:dev AS build
+
+WORKDIR /airbyte
+ENV APPLICATION destination-opensearch
+
+COPY build/distributions/${APPLICATION}*.tar ${APPLICATION}.tar
+
+RUN tar xf ${APPLICATION}.tar --strip-components=1 && rm -rf ${APPLICATION}.tar
+
```
[the remark this answers is not in the document]
Comment (a1tair6) [reply airbyte-integrations/connectors/destination-opensearch/Dockerfile]: I fixed it.

File: docs/integrations/destinations/opensearch.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# OpenSearch
+
+## Sync overview
+
+### Output schema
+
```
[the remark this answers is not in the document]
Comment (a1tair6) [reply docs/integrations/destinations/opensearch.md:6]: I added it.

**Comment by a1tair6** [comment]
> I tested the integration test and its not working. Can you confirm you can run locally? `./gradlew airbyte-integrations:connectors:source-opensearch:integrationTest`
> 
> Also it's missing the icon and the metadata.yaml file inside the connector folder. Take a look in other Java connectors to reference.

I'm sorry. I made a mistake in writing the README.

Can you test this?
`./gradlew airbyte-integrations:connectors:source-opensearch:integrationTestJava`

**Comment by marcosmarxm** [comment]
Sorry the delay to return to your contribution @jay-choe but it looks the integrationTest aren't working.
You can check https://github.com/airbytehq/airbyte/pull/28861#issuecomment-1658682700 where Airbyte's CI is failing to run the proper tests. I tried to fix a few lines but it looks there are more to be fixed. Let me know if you want to continue the contribution for this connector.

**Comment by marcosmarxm** [comment]
@jay-choe did you check my last message?

**Comment by a1tair6** [comment]
> @jay-choe did you check my last message?

I'm sorry for the delayed response. I'll take a look.

**Comment by jay-choe** [comment]
@marcosmarxm Sorry for delayed response, and we want to continue this pull request.
We checked our local machine that all tests we wrote run successfully.
And i don't know why yaml validation or failed process happened.
Is there any lint issue or something?

In adddition, as you said, 'more to be fixed' can you elaborate this?
We want your guide and follow.

**Comment by marcosmarxm** [comment]
> @marcosmarxm Sorry for delayed response, and we want to continue this pull request. We checked our local machine that all tests we wrote run successfully. And i don't know why yaml validation or failed process happened. Is there any lint issue or something?
> 
> In adddition, as you said, 'more to be fixed' can you elaborate this? We want your guide and follow.

Can you confirm the `./gradlew airbyte-integrations:connectors:source-opensearch:integrationTest` is working for you?

**Comment by yugwon-oh** [comment]
@marcosmarxm 
We need to develop based on Java and test it using `./gradlew airbyte-integrations:connectors:source-opensearch:integrationTestJava`. 
Is `./gradlew airbyte-integrations:connectors:source-opensearch:integrationTest` for Python integration testing?
I'm curious if it's also possible to test with this.

**Comment by a1tair6** [comment]
@marcosmarxm
We need to develop based on Java and test it using ./gradlew airbyte-integrations:connectors:source-opensearch:integrationTestJava.
Is ./gradlew airbyte-integrations:connectors:source-opensearch:integrationTest for Python integration testing?
I'm curious if it's also possible to test with this.

**Comment by marcosmarxm** [comment]
No @jay-choe integration tests are both for Java and Python connectors.

**Comment by jay-choe** [comment]
@marcosmarxm Sorry For delayed response. We have been really busy thesedays.
As you mensionsed, integration tests failed and there is a gradle version issued as the error message says.
We rasied this PR long time ago and is there any change we can follow? Or can you give me a hand?
image

**Comment by marcosmarxm** [comment]
@jay-choe at the moment I don't have time. At the moment all efforts are dedicated to review Hackathon contributions.

**Comment by jay-choe** [comment]
@marcosmarxm I understand. 
Feel free to review this :)
After we rebase it, i guess it passes all test.

**Comment by a1tair6** [comment]
@marcosmarxm Could you please review this?

**Comment by marcosmarxm** [comment]
Hello! Thanks for your contribution. We’ve noticed it’s been a while since you received an update. After internal discussions, the team has decided not to accept new contributions for Java sources and destinations at the moment. Airbyte is currently revamping the shared core destinations codebase and plans to release a new Java destinations CDK in 2024.
The team is working on improving general usability, speed through asynchronous loading, and Typing and Deduplication (Destinations V2). Accepting contributions in the current state has led to confusion among users. In the meantime, feel free to build and host the images independently. Share them with users interested in your connector until the official Java CDK is released and ready to accept new connectors. Thanks for your understanding and feel free to reach me directly in Slack to discuss the topic if you needed.

**Pull request closed without merging**